explainer.md

```diff
@@ -186,225 +181,236 @@ https://user-images.githubusercontent.com/93594/141100217-ba1fa157-cd79-4a9d-b3b
+1. Add/remove offered elements according to `setElement` calls.
+1. If at least one element remains offered, a transition can go ahead.
 
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/79): Do we need `createTransitionElement`? It could be done via adding CSS properties, but it might be clumsy if the developer is going to immediately remove the properties afterwards.
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/80): How does the outgoing page offer just the root for transition?
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/81): When is the `performTransition` callback called?
+As a result of the above, if an element is offered via CSS and `setElement`, then `setElement` wins. If there are multiple calls to `setElement` for the same element, the last wins.
 
-## Defining the animation
+Gathering offered elements _after_ the dispatch of `pagehide` means the developer can use a mix of the CSS and JS methods. For instance, the developer could set a class on the root element depending on `event.nextUrl`, which changes the offered elements.
 
-How will Page-B define the animation?
+## Page-B
 
-### Default animation
+Page-B must also offer elements to use for a transition to happen. Page-B also controls the animation of the transition.
 
-The potential default animations setup for the different cases are as follows. Note all of these can be overridden by the developer:
+### Via CSS
 
-- If an element exists in Page-A only (exit animation), a fade animation takes its container from opacity 1 to 0.
-- If an element exists in Page-B only (entry animation), a fade animation takes its container from opacity 0 to 1.
-- If an element exists in both Page-A and Page-B, and both are 'container and content', an animation takes the container from Page-A styles to Page-B styles (which will include the transform used for positioning), while cross-fading the two images.
-- If an element exists in both Page-A and Page-B, and neither are 'container and child', a transform animation takes its container from Page-A size/transform to Page-B size/transform, while cross-fading the two images.
+#### Offering Page-B elements
 
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/82): What if the Page-A element is 'container and child' but the Page-B element is 'single image'?
+Page-B elements are offered for the transition using the same mechanism as in Page-A, `page-transition-tag`.
 
-Because the images are sized to 100% of the container, the images will also change size throughout the transition. How these are scaled can be changed using regular CSS features like `object-fit`.
+Elements from Page-A and Page-B that have the same tag are merged into one transition elements as documented in "Mixing in elements from Page-B and associating them with transition elements from Page-A". If multiple elements have the same tag, they're associated in DOM order.
```

Comment:
Clarifying: if I have 3 elements with tag 'foo' on the old page and 5 on the new page, then the last 2 on the new page are not transitioned?

Also random thought: I'm not sure how this works with shadow dom. I can presumably have shadow dom elements transition here, I guess they also have a well defined DOM order? But is that the same order as the rendered order?

Things like querySelectorAll don't seem to specify the order in which they return the element. querySelector says "first" but I guess all of this is also DOM order?

Reply:
> Clarifying: if I have 3 elements with tag 'foo' on the old page and 5 on the new page, then the last 2 on the new page are not transitioned?

I don't think I explained this well. The first three would be matched, but the last 2 would be unmatched but still part of the transition (entry transitions).

Reply:
The ordering stuff is interesting. I need to think about this more. Especially if we're still thinking of allowing elements in iframes to be shared elements.